presto-bytecode/src/main/java/com/facebook/presto/bytecode/MethodDefinition.java

```diff
@@ -70,6 +72,14 @@ public MethodDefinition(
             Iterable<Parameter> parameters
     )
     {
+        // Code generation could failed if parameter count is 255 or more. Given that fixing this directly might be difficult,
+        // handle it by capping the maximum number of arguments to 250.
+        int parameterCount = Iterables.size(parameters);
```

Comment:
Why 254 and not 255? Is the "this" parameter not included? If so, we could make this
```java
int parameterLimit = access.contains(STATIC) ? 255 : 254;